test/any/web_catalog/api-extractor-test.es6.js

```diff
@@ -0,0 +1,331 @@
+      extractor.arrayMerge_(array1, array2, array3);
+      expect(array1).toEqual(['a', 'b', 'f']);
+    });
+  });
+
+  describe('Method setMinus', function() {
+    it('correctly find set minus between two arrays', function() {
+      let array1 = ['a', 'b', 'c'];
+      let array2 = ['b', 'd'];
+      expect(extractor.setMinus_(array1, array2)).toEqual(['a', 'c']);
+      let array3 = ['a', 'b', 'c'];
+      let array4 = ['b', 'd', 'a', 'c', 'e'];
+      expect(extractor.setMinus_(array3, array4)).toEqual([]);
+    });
+    it('clean up for string.', function() {
+      let array1 = ['$a$', '$b$', '$c$', '+d+'];
+      let array2 = ['b', 'd'];
+      expect(extractor.setMinus_(array1, array2)).toEqual(['a', 'c']);
+      let array3 = ['a', 'b', '$c$', '+e+'];
+      let array4 = ['b', 'd', '$a$', 'c', 'e'];
+      expect(extractor.setMinus_(array3, array4)).toEqual([]);
+    });
+  });
+
+  describe('Method getFunctionBuildInProperties', function() {
+    let og = objectGraph.fromJSON({
+      'data': {
+        '10000': {
+          'Function': 10001,
+        },
+        '10001': {
+          'prototype': 10002,
+        },
+        '10002': {
+          'caller': 3,
+          'arguments': 6,
+          'length': 6,
+          '+toString+': 6,
+        },
+      },
+      types,
+      'root': 10000,
+    });
+    it('obtain Function build-in properties correctly', function() {
+      extractor.getFunctionBuildInProperties_(og);
+      expect(extractor.buildInFunctionProperties.sort())
+        .toEqual(['prototype', 'caller', 'arguments', 'length', 'toString']
+        .sort());
+    });
+    it('clean up string for properties', function() {
+      extractor.getFunctionBuildInProperties_(og);
+      expect(extractor.buildInFunctionProperties).toContain('toString');
+    });
+  });
+
+  describe('Method getObjectBuildInProperties', function() {
+    let og = objectGraph.fromJSON({
+      'data': {
+        '10000': {
+          'Object': 10001,
+        },
+        '10001': {
+          'prototype': 10002,
+        },
+        '10002': {
+          '+toString+': 3,
+          '+valueOf+': 6,
+          '+constructor': 6,
+        },
+      },
+      types,
+      'root': 10000,
+    });
+    it('obtain cleaned up Object build-in properties correctly', function() {
+      extractor.getObjectBuildInProperties_(og);
+      expect(extractor.buildInObjectProperties.sort())
+        .toEqual(['toString', 'valueOf', 'constructor']
+        .sort());
+    });
+  });
+
+  describe('Method getClassName', function() {
+    it('obtain class name of object\'s proto is its class', function() {
+      let og = objectGraph.fromJSON({
+        'data': {
+          '10010': {
+            'instanceProperty': 3,
+          },
+          '10020': {
+            '+constructor+': 10030,
+          },
+        },
+        'protos': {
+          '10010': '10020',
+        },
+        'functions': {
+          '10030': 'class',
```

Comment:
nit: here and elsewhere: `class` is reserved. A dummy class-name would be more realistic; the Chrome team seems to like `Frobinator` as to mean "canonical thing".